src/System.Management.Automation/namespaces/FileSystemProvider.cs

```diff
@@ -3104,22 +3112,31 @@ private void RemoveDirectoryInfoItem(DirectoryInfo directory, bool recurse, bool
                 continueRemoval = ShouldProcess(directory.FullName, action);
             }
 
-            if (directory.Attributes.HasFlag(FileAttributes.ReparsePoint))
+            if (InternalSymbolicLinkLinkCodeMethods.IsReparsePointLikeSymlink(directory))
             {
+                void WriteErrorHelper(Exception exception)
```

Comment:
This method is probably generally applicable enough that it could just be a private method on the class.

Also, we can just call it `WriteError` -- that's what overloads are for. Methods that are called `<...>Helper` are rarely descriptive or well-scoped.

Reply:
It is an _internal_ function to exclude code duplication.